examples/raylib/raylib.c

```diff
@@ -0,0 +1,271 @@
+{
+    struct Color color = set_color(objgr->color);
+    tmx_object *head = objgr->head;
+    Rectangle rect;
+    Vector2 *vectorPoints; /* can be fixed array to avoid malloc*/
+    double **raw_points;
+    int x, y, i;
+
+
+    while (head) {
+        if (head->visible) {
+            if (head->obj_type == OT_SQUARE) {
+                rect.x = head->x;
+                rect.y = head->y;
+                rect.width = head->width;
+                rect.height = head->height;
+                DrawRectangleLines(rect.x -player.x, rect.y -player.y, rect.width, rect.height, color);
+            } else if (head->obj_type  == OT_POLYGON) {
+                draw_polygon(head->content.shape->points, head->x -player.x, head->y -player.y, head->content.shape->points_len, color);
+            } else if (head->obj_type == OT_POLYLINE) {
+
+                /* You have to make Vector2* out of double** data points */
+                vectorPoints = malloc (head->content.shape->points_len * sizeof (Vector2));
+                for (i = 0; i < head->content.shape->points_len; i++) {
+                    raw_points = head->content.shape->points;
+                    x = head->x + raw_points[i][0];
+                    y = head->y + raw_points[i][1];
+                    vectorPoints->x = x -player.x;
+                    vectorPoints->y = y -player.y;
+                    vectorPoints++;
+                }
+                vectorPoints -= i;
+                DrawPolyExLines(vectorPoints, head->content.shape->points_len, color);
+
+                /* Or you can use this */
+                //draw_polyline(head->content.shape->points, head->x -player.x, head->y -player.y, head->content.shape->points_len, color);
+            } else if (head->obj_type == OT_TEXT) {
+                DrawText(head->content.text->text, head->x -player.x, head->y -player.y, head->content.text->pixelsize, color);
+            } else if (head->obj_type == OT_ELLIPSE) {
+                /* Ellipse function */
+            }
+        }
+        head = head->next;
+    }
+}
+
+unsigned int gid_clear_flags(unsigned int gid)
+{
+    return gid & TMX_FLIP_BITS_REMOVAL;
+}
+
+void draw_layer(tmx_map *map, tmx_layer *layer)
+{
+    int i, j;
+    unsigned int gid;
+    tmx_tileset *ts;
+    tmx_image *im;
+    Rectangle srcrect;
+    Vector2 dstrect;
+    Texture* tileset;
+    for (i=0; i<map->height; i++) {
+        for (j=0; j<map->width; j++) {
+            gid = gid_clear_flags(layer->content.gids[(i*map->width)+j]);
+            if (map->tiles[gid] != NULL) {
+                ts = map->tiles[gid]->tileset;
+                im = map->tiles[gid]->image;
+                srcrect.x = map->tiles[gid]->ul_x;
+                srcrect.y = map->tiles[gid]->ul_y;
+                srcrect.width = ts->tile_width;
+                srcrect.height = ts->tile_height;
+                dstrect.x = j*ts->tile_width;
+                dstrect.y = i*ts->tile_height;
+                  dstrect.x -= player.x;
```

Comment:
I know it's clear that all these `-= player.x` lines change render positions so it looks like camera follows the player.
Wrote this because I have no idea whether it's a common practice or quite an uncommon way to make camera to follow the player.